NumericOutcome.md

```diff
@@ -0,0 +1,453 @@
+        (lastImportantDigit + 1).until(base).map { lastDigit => // Note that this range excludes lastImportantDigit and base
+          fixedDigits :+ lastDigit
+        }
+    }
+
+      if (digits.last == 0) { // Row Optimization
+        digits.init +: fromFront.drop(base - 1) // Note init drops the last digit and drop(base-1) drops the fist row
+      } else {
+      digits +: fromFront // Add outlier
+    }
+  }
+}
+
+def backGroupings(
+    digits: Vector[Int], // The unique digits of the range's end
+    base: Int): Vector[Vector[Int]] = {
+  if (digits.tail.forall(_ == base - 1)) { // Grouping Optimization
+    Vector(Vector(digits.head))
+  } else {
+    // Here we compute the back groupings in reverse so as to use the same iteration as in front groupings
+    val fromBack = digits.reverse.zipWithIndex.init.flatMap { // Note the flatMap collapses the rows of the grouping 
+      case (lastImportantDigit, unimportantDigits) =>
+        val fixedDigits = digits.dropRight(unimportantDigits + 1)
+        0.until(lastImportantDigit).reverse.toVector.map { // Note that this range excludes lastImportantDigit
+          lastDigit =>
+            fixedDigits :+ lastDigit
+        }
+    }
+
+    if (uniqueEndDigits.head == base - 1) { // Row Optimization
+      fromBack.drop(base - 1).reverse :+ digits.init // Note init drops the last digit and drop(base-1) drops the last row
+    } else {
+      fromBack.reverse :+ digits // Add outlier
+    }
+  }
+}
+
+def middleGrouping(
+    firstDigitStart: Int, // The first unique digit of the range's start
+    firstDigitEnd: Int): Vector[Vector[Int]] = { // The first unique digit of the range's end
+  (firstDigitStart + 1).until(firstDigitEnd).toVector.map { firstDigit => // Note that this range excludes firstDigitEnd
+    Vector(firstDigit)
+  }
+}
+```
+
+Finally we are able to use all of these pieces to compress a range to an approximately minimal number of outcomes (by ignoring digits).
+
+```scala
+def groupByIgnoringDigits(start: Long, end: Long, base: Int, numDigits: Int): Vector[Vector[Int]] = {
+    val (prefixDigits, startDigits, endDigits) = separatePrefix(start, end, base, numDigits)
+    
+    if (start == end) { // Special Case: Range Length 1
+        Vector(startDigits)
+    } else if (startDigits.forall(_ == 0) && endDigits.forall(_ == base - 1)) { // Total Optimization
+        Vector(prefixDigits)
+    } else if (prefix.length == numDigits - 1) { // Special Case: Front Grouping = Back Grouping
+        startDigits.last.to(endDigits.last).toVector.map { lastDigit =>
+            prefixDigits :+ lastDigit
+        }
+    } else {
+      val front = frontGroupings(startDigits, base)
+      val middle = middleGrouping(startDigits.head, endDigits.head)
+      val back = backGroupings(endDigits, base)
+
+      val groupings = front ++ middle ++ back
+
+      groupings.map { digits =>
+        prefixDigits ++ digits
+      }
+    }
+}
+```
+
+## General Payout Curves
```

Comment:
If you want to have something merge in the not so distant future, I would take that in a different PR. It's going to be difficult to give you feedback on that without actually trying to implement it myself I feel.